# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: toolkit/components/antitracking/StoragePrincipalHelper.h
# Commit: ff5ea82f6090
# Full Hash: ff5ea82f6090a4a1645a1a32749a4ae9102f015b
# Author: Andrea Marchesini <amarchesini@mozilla.com>
# Date: 2025-04-17 16:30:04
# Regressor Bug: 1957355
# File Overlap Count: 1
# Description:
#   Bug 1957355 - No storage access for resources injected by content-scripts, r=timhuang,necko-reviewers,devtools-reviewers,anti-tracking-reviewers,nchevobbe,kershaw
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D244204
# ==============================================================================

diff -r 04eed8759454 -r ff5ea82f6090 toolkit/components/antitracking/StoragePrincipalHelper.h
--- a/toolkit/components/antitracking/StoragePrincipalHelper.h	Thu Apr 17 10:56:59 2025 +0000
+++ b/toolkit/components/antitracking/StoragePrincipalHelper.h	Thu Apr 17 10:56:59 2025 +0000
@@ -359,6 +359,11 @@
   // "(https,foo.com)", true -> "(https,foo.com,f)"
   static void UpdatePartitionKeyWithForeignAncestorBit(
       nsAString& aKey, bool aForeignByAncestorContext);
+
+  // Generate a partition-key for an expanded principal to be used for any
+  // injected resources by add-on content scripts.
+  static nsString PartitionKeyForExpandedPrincipal(
+      nsIPrincipal* aExpandedPrincipal);
 };
 
 }  // namespace mozilla
